### PR TITLE
Refactors Camera Damage and Disabling

### PR DESCRIPTION
**Pull request by Fox-McCloud**
This was supposed to be a simple PR, but no, nothing is ever easy.

Ports over cameras to the object damage system and alters how you enable/disable them.

Actual changes
- Can deactivate and destroy cameras with objects that have a force of above 12
  - Repairing a camera is done by cutting/mending its wire for power
- Reduces camera wire count from 6 to 2
  - Current wires are now camera-focus and power
- Removes arbitrary limit on not being able to use hand labelers on reagent containers

:cl: Fox McCloud
fix: Fixes hand labelers not being able to label storage objects
add: Cameras can now be destroyed with weapons that have a force of 12 or greater
tweak: Cameras have 2 wires as opposed to 6; camera focus and power wires still remain
tweak: You can use hand labelers on reagent containers now
/:cl:

### CODE REVIEW

**Comment by Bram961** [comment]
Not sure if I like cameras being in the damage system for 12 damage, that seems like quite the nerf for the AI

**Comment by Fox-McCloud** [comment]
I do hope you realize that anyone with wirecutters and a screwdriver can effectively do the same exact thing as a bonking it a bunch of times with a 12 damage item can do.

**Comment by Regen1** [comment]
I don't like the change that makes the camera's more "stupid"

I've seen plenty of people not knowing about the alarm wire, and it usually takes *some* time to cut all wires and leaving the alarm wire.

The wires adds to the atmosphere of ss13, removing it seems unnecessary and arbitrary.

**Comment by Fox-McCloud** [comment]
The wires themselves are arbitrary and unnecessary; they add an additional layer of complexity that doesn't really accomplish much, with the **sole** exception of signalers.

Balancing around "well, I know a lot of people who don't know about X" isn't a good design perspective. Intuitive gameplay is preferable of additional complexity (for the sake of it, without any depth). 

The assertion that it takes "time" is also pretty trivial; you're talking a time difference of less than 2 seconds.

**Comment by Regen1** [comment]
>The wires themselves are arbitrary and unnecessary; they add an additional layer of complexity that doesn't really accomplish much, with the sole exception of signalers.

We could say the same about door wires, but I disagree completely.

Yes, it adds an additional layer of complexity, but that has a charm to it. If anything it just makes it feel like you're more in the game, this might be subjective, but I don't think it's your place, or mine, to say either or. 

>Balancing around "well, I know a lot of people who don't know about X" isn't a good design perspective. Intuitive gameplay is preferable of additional complexity (for the sake of it, without any depth).

I'm not really looking for balance, I'm looking for what makes the game more enjoyable, fun and unique.

**Comment by Regen1** [comment]
>The purpose of making the game more balance is to make it more enjoyable for everyone; sometimes that means simplifying existing behaviors to make them more easy to understand or approachable. In this case, complexity is being reduced to make the system more approachable, easy to understand, and reducing an additional layer of "fiddliness" that doesn't really serve any purpose.

My issue is that I feel that if we do this too much, our server's just going to loose more and more "ss13 feel" and be more "standard game".
I don't like removing it, but I understand your point.

I'd much rather add more complexity to it (So it's inline with doors, like your example), other than "beat arbitrary force value to remove a camera", because at least before this PR, you were required at least two tools to remove them, 3 if you want to do it safely.

**Comment by tigercat2000** [comment]
I like the damage, I don't like removing the wires. It's a nice minigame. There's not really any reason for vendors to be anything other than a multitool to enable contraband either.

**Comment by alexkar598** [comment]
Ss13 is meant to be a complex game,removing features removes from the charm of the game,lets remove surgery steps for a simple click with bone gel for 2 minutes instead

**Comment by Spacemanspark** [comment]
I'd prefer to keep the wires.

**Comment by Spacemanspark** [comment]
The camera alarm is a thing, and I've used it several times.
You'll really notice it when a place gets EMP'd.

**Comment by Fruerlund** [comment]
Keep the wires, I like the damage thing.

**Comment by Spartan6** [comment]
Missing Balance tag...

**Comment by Fox-McCloud** [comment]
` I don't like removing the wires. It's a nice minigame. There's not really any reason for vendors to be anything other than a multitool to enable contraband either.`

Not really. The wires add additional depth to the vending machine; it has more than just contraband--it can throw items, shock you, or disables ID access; it's not a simplistic interaction, like the camera is.

**Comment by Regen1** [comment]
Technically this is also a balance change, considering it now only takes a force 12 or above weapon, as opposed to 2-3 tools

**Comment by Fruerlund** [comment]
> Wires on a camera, of which there's really only 2 functions? That's not extra depth; you're just making a player jump through hoops, at that stage.

Not really. As stated in other comments some players specifically use these "hoops". Besides, if its such a big problem to keep redundant wires. Just remove all the wires that do not have any relevant action, meaining cameras would only have two wires (IFRC), instead of four.

Also because it has been, in the past, doesn't mean it should'nt be in the future either.

**Comment by RyanSmake** [comment]
> The purpose of making the game more balance is to make it more enjoyable for everyone
This is just so untrue.
Also camera wires are inline with how almost every other machine works where familiar functionality is needed. Also now you can disable a camera just by smashing it a few times with a glass shard, and I disapprove of that. It completely changes how the camera can be deactivated, in fact it makes it so easy, there is just no reason to grab a screwdriver anymore.

**Comment by Rurik123** [comment]
> Camera wires are inline with how almost every other machine works where familiar functionality is needed. Also now you can disable a camera just by smashing it a few times with a glass shard, and I disapprove of that. It completely changes how the camera can be deactivated, in fact it makes it so easy, there is just no reason to grab a screwdriver anymore.

I think you overstate how much  _easier_ this will make things. Grabbing a screwdriver and multitool will always be preferable.

**Comment by Plotron** [comment]
I expect to find wires in a camera just like in any other machine and I am in favor of cutting down on bloat, so I'd go with removing the redundant wires and just keeping the minimum to preserve current functionality. As such, I agree with @Fruerlund.

**Comment by FalseIncarnate** [comment]
I'm not gonna even touch the camera stuff here... but what's with the reagent container change?

You can already label them with a pen, and that seems like such a random change to bundle into this PR. Mind explaining the reasoning on why that was included here?

**Comment by Fox-McCloud** [comment]
>You can already label them with a pen, and that seems like such a random change to bundle into this PR. Mind explaining the reasoning on why that was included here?

Yeah, it's actually rather silly. In order to get sections of this code working, I had to pretty much ensure a lot of things returned the value of `attackby` to parent. This naturally meant that I had to touch `/obj/item/storage`, which made some problems obvious, like the hand labeler, which necessitated me changing the hand labeler's code, directly too--upon which I noticed some bad bode and inconsistencies...then I noticed, that for whatever reason, explicitly, reagent containers were snowflake exempted, soooo just went ahead and removed that in the cleanup.

**Comment by Ziiro** [comment]
Instead of removing the wires, give them function. Shock wire, night vision wire.

**Comment by theColdflame** [comment]
imo keep the wires as they are, make smashing trigger the alarm. adds different methods of tackling the problem of a camera with distinct pros and cons

**Comment by Fox-McCloud** [comment]
Re-added wires, though they're simplified.

File: code/datums/wires/camera.dm [hunk 1]
```diff
@@ -3,48 +3,35 @@
...
 	holder_type = /obj/machinery/camera
-	wire_count = 6
+	wire_count = 3
 
 /datum/wires/camera/get_status()
 	. = ..()
 	var/obj/machinery/camera/C = holder
 	. += "The focus light is [(C.view_range == initial(C.view_range)) ? "on" : "off"]."
 	. += "The power link light is [C.can_use() ? "on" : "off"]."
-	. += "The camera light is [C.light_disabled ? "off" : "on"]."
```
Comment (Fox-McCloud) [inline code/datums/wires/camera.dm]: This did nothing.

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -295,15 +291,10 @@
...
-	alarm_on = 1
+/obj/machinery/camera/proc/triggerCameraAlarm()
 	motion_alarm.triggerAlarm(loc, src)
 
 /obj/machinery/camera/proc/cancelCameraAlarm()
-	if(wires.IsIndexCut(CAMERA_WIRE_ALARM))
-		return
-
-	alarm_on = 0
 	motion_alarm.clearAlarm(loc, src)
```
Comment (Crazylemon64) [inline code/game/machinery/camera/camera.dm]: Are you sure this should be `motion_alarm` and not `camera_alarm`? The latter is unused anywhere else throughout the codebase, besides for borg registration.

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -295,15 +291,10 @@
 			O.reset_perspective(null)
 			to_chat(O, "The screen bursts into static.")
 
-/obj/machinery/camera/proc/triggerCameraAlarm(var/duration = 0)
-	alarm_on = 1
+/obj/machinery/camera/proc/triggerCameraAlarm()
 	motion_alarm.triggerAlarm(loc, src)
```
Comment (Crazylemon64) [inline code/game/machinery/camera/camera.dm]: Same here

File: code/game/machinery/camera/camera.dm (hunk 2 above)
Comment (Crazylemon64) [reply code/game/machinery/camera/camera.dm]: This was changed 2 years ago by Tigercat's /tg/ cameranet port - and should probably be changed back

File: code/game/machinery/camera/camera.dm [hunk 4]
```diff
@@ -71,144 +71,120 @@
...
+	if(isscrewdriver(I))
 		panel_open = !panel_open
-		user.visible_message("<span class='warning'>[user] screws the camera's panel [panel_open ? "open" : "closed"]!</span>",
-		"<span class='notice'>You screw the camera's panel [panel_open ? "open" : "closed"].</span>")
-		playsound(src.loc, W.usesound, 50, 1)
+		to_chat(user, "<span class='notice'>You screw the camera's panel [panel_open ? "open" : "closed"].</span>")
+		playsound(loc, I.usesound, 50, 1)
 
-	else if((istype(W, /obj/item/wirecutters) || istype(W, /obj/item/multitool)) && panel_open)
+	else if((iswirecutter(I) || ismultitool(I) && panel_open))
```
[the remark this answers is not in the document]
Comment (Fox-McCloud) [reply code/game/machinery/camera/camera.dm]: Oops, yeah, my mistake here.

**Pull request merged**